cli/commons/src/Microsoft.Kiota.Cli.Commons/IO/JsonOutputFormatter.cs

```diff
@@ -0,0 +1,127 @@
+        {
+            var result = ProcessJson(content, jsonOptions.OutputIndented);
+            _ansiConsole.WriteLine(result);
+        }
+        else
+        {
+            _ansiConsole.WriteLine(content);
+        }
+    }
+
+    /// <inheritdoc />
+    public void WriteOutput(Stream content, IOutputFormatterOptions options)
+    {
+        using var reader = new StreamReader(content);
+        var strContent = reader.ReadToEnd();
+        if (options is JsonOutputFormatterOptions jsonOptions && jsonOptions.OutputIndented)
+        {
+            var result = ProcessJson(strContent, jsonOptions.OutputIndented);
+            _ansiConsole.WriteLine(result);
+        }
+        else
+        {
+            _ansiConsole.WriteLine(strContent);
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task WriteOutputAsync(Stream content, IOutputFormatterOptions options, CancellationToken cancellationToken = default)
+    {
+        string resultStr;
+
+        if (options is JsonOutputFormatterOptions jsonOptions && jsonOptions.OutputIndented)
+        {
+            using var result = await ProcessJsonAsync(content, jsonOptions.OutputIndented, cancellationToken);
+            using var r = new StreamReader(result);
+            resultStr = await r.ReadToEndAsync();
+        }
+        else
+        {
+            using var reader = new StreamReader(content);
+            resultStr = await reader.ReadToEndAsync();
+        }
+
+        _ansiConsole.WriteLine(resultStr);
+    }
+
+    /// <summary>
+    /// Given a JSON input string, returns a processed JSON string with optional indentation
+    /// </summary>
+    /// <param name="input">JSON input string</param>
+    /// <param name="indent">Whether to return indented output</param>
+    private static string ProcessJson(string input, bool indent = true)
+    {
+        var result = input;
+        try
+        {
+            var jsonDoc = JsonDocument.Parse(input);
+            result = JsonSerializer.Serialize(jsonDoc, options: new() { WriteIndented = indent });
+        }
+        catch (JsonException)
+        {
+        }
+
+        return result;
+    }
+
+    /// <summary>
+    /// Given a JSON input stream, returns a processed JSON stream with optional indentation
+    /// </summary>
+    /// <param name="input">JSON input stream</param>
+    /// <param name="indent">Whether to return indented output</param>
+    /// <param name="cancellationToken">The cancellation token</param>
+    private static async Task<Stream> ProcessJsonAsync(Stream input, bool indent = true, CancellationToken cancellationToken = default)
+    {
+        var buffer = new byte[input.Length];
+        await input.ReadAsync(buffer, cancellationToken);
+
+        try
+        {
+            using var cachedStream = new MemoryStream(buffer);
+            var jsonDoc = await JsonDocument.ParseAsync(cachedStream, default, cancellationToken);
```

Comment:
Any reason we can't read the `input` stream directly here rather than copying it to second stream?
If things go wrong and an exception thrown, we might be able to rewind it and return something rather that the common case always involving duplication/copying of data.

Reply:
Because of the possibility on streams that don't support seeking. These streams can only be read once.

Reply:
That's a good catch. Thank you. I've made an update to the copy logic. It only happens when the input stream isn't seekable